### PR TITLE
Enable Viewing of Projects Across Multiple Rounds

### PR DESCRIPTION
**Pull request by nijoe1**
### Header Refactoring:

**Introduced dropdown functionality for each navigation link.**

- useFilter Refactoring:

   - Enhanced the useFilter hook to support querying projects by specific rounds.

- Projects Link Enhancement:

   - Added dropdown items under the Projects link to allow users to view projects from previous rounds.
   - Implemented the same dropdown functionality for the mobile view

### CODE REVIEW

**Comment by nijoe1** [comment]
fixes PAR-304

**Review by bhargavaparoksham — DISMISSED** [review]

File: src/components/hooks/useAnchorPosition.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from "react";
+
+const useAnchorPosition = (anchorEl: any) => {
+  const [position, setPosition] = useState<
```
Comment (carlbarrdahl) [inline src/components/hooks/useAnchorPosition.ts:5]: See if you can use React-Use for this:
https://github.com/streamich/react-use

File: src/components/hooks/useIsMobile.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from "react";
+
+const useIsMobile = (
+  // Defaults at md: 768px
+  breakpoint = 768,
```
Comment (carlbarrdahl) [inline src/components/hooks/useIsMobile.ts:6]: Same here, there's a hooks for this with React-Use:
https://github.com/streamich/react-use/blob/master/docs/createBreakpoint.md

File: src/layouts/DefaultLayout.tsx [hunk 3]
```diff
@@ -29,21 +36,25 @@ export const Layout = ({ children, ...props }: Props) => {
   //     children: "Stats",
   //   });
   // }
+  const dropdownItems = undefined;
```
Comment (carlbarrdahl) [inline src/layouts/DefaultLayout.tsx]: Could this be optional in the navLinks instead?

**Review by carlbarrdahl — DISMISSED** [review]
Good job Nick!

File: src/components/hooks/useActiveNavLink.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      link.dropdownItems.some((item) => asPath.includes(item.key));
+
+    // Check if the current link's href is also used in a dropdown
+    const isInDropdown = dropdownHrefMap.has(link.href);
+
+    // Determine if the current link should be active
+    return isDropdownActive
+      ? true // Mark the dropdown link as active if any of its items are active
+      : asPath.startsWith(link.href) && !isInDropdown && !link.dropdownItems; // Only mark non-dropdown links as active if they are not in the dropdown map
+  };
```
Comment (carlbarrdahl) [inline src/components/hooks/useActiveNavLink.ts:43]: Can be shortened:
```tsx
return isDropDownActive && asPath.startsWith(link.href) && !isInDropdown && !link.dropdownItems
```

File: src/components/hooks/useActiveNavLink.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    });
+    return map;
+  }, [asPath, navLinks]);
+
+  // Step 2: Determine if a link is active
+  const isLinkActive = (link: NavLinkType): boolean => {
+    // Check if the link has dropdown items and if any item.key of them match the current path
+    const isDropdownActive =
+      link.dropdownItems &&
+      link.dropdownItems.some((item) => asPath.includes(item.key));
```
Comment (carlbarrdahl) [inline src/components/hooks/useActiveNavLink.ts:34]: Can use optional chaining:
```tsx
const isDropdownActive = link?.dropdownItems.some(...
```

File: src/components/hooks/useIsMobile.ts (hunk 2 above)
Comment (carlbarrdahl) [reply src/components/hooks/useIsMobile.ts:6]: This hook seem unnecessary and can be done like this:

```tsx
const isMobile = useBreakpoint() === "S"
```

File: src/features/ballot/components/BallotOverview.tsx [hunk 6]
```diff
@@ -51,10 +51,10 @@ function BallotOverview() {
           size="sm"
           variant="secondary"
           onClick={download.downloadMetadata}
-          isLoading={download.isLoading || download.fetched}
-          disabled={download.isLoading}
+          isLoading={download.isLoading || download.count === 0}
+          disabled={download.isLoading || download.count === 0}
         >
-          Download
+          {`Download projects`}
```
Comment (carlbarrdahl) [inline src/features/ballot/components/BallotOverview.tsx:57]: Remove the brackets and quotes to simply: `Download projects`

File: src/features/ballot/components/BallotOverview.tsx [hunk 7]
```diff
@@ -51,10 +51,10 @@ function BallotOverview() {
           size="sm"
           variant="secondary"
           onClick={download.downloadMetadata}
-          isLoading={download.isLoading || download.fetched}
-          disabled={download.isLoading}
+          isLoading={download.isLoading || download.count === 0}
+          disabled={download.isLoading || download.count === 0}
```
Comment (carlbarrdahl) [inline src/features/ballot/components/BallotOverview.tsx:55]:
```tsx
download.isLoading || !download.count
```

**Review by carlbarrdahl — APPROVED** [review]
Looks good! Some comments regarding shorter ways to write certain things.

**Review by carlbarrdahl — APPROVED** [review]

**Pull request merged**